airflow/contrib/operators/sftp_operator.py

```diff
@@ -77,13 +81,19 @@ def __init__(self,
     def execute(self, context):
         file_msg = None
         try:
-            if self.ssh_conn_id and not self.ssh_hook:
-                self.ssh_hook = SSHHook(ssh_conn_id=self.ssh_conn_id)
+            if self.ssh_conn_id:
+                if self.ssh_hook:
+                    self.log.info("ssh_conn_id is ignored when ssh_hook is provided.")
+                else:
+                    self.ssh_hook = SSHHook(ssh_conn_id=self.ssh_conn_id)
```

Comment:
why don't we set timeout for this hook but the one in SSHOperator?

Reply:
I have noticed this as well. I'm following the existing implementation.

`timeout` was added for `SSHOperator` in PR https://github.com/apache/incubator-airflow/pull/3666 . I'm not sure if there is any good reason for which `timeout` was added for `SSHOperator` but not added for the `SSHHook` in `SFTPOperator`. Giving I'm not confident about the reason behind, I don't intend to make change on this in this PR as well.